### PR TITLE
Cleaning and documenting cnf stream

### PR DESCRIPTION
**Pull request by HanielB**
Moreover changes assertClause to return a bool, which is gonna be used by the proof cnf stream.

### CODE REVIEW

File: src/prop/cnf_stream.h [hunk 1]
```diff
@@ -54,7 +59,122 @@ class CnfStream {
...
+   * Constructs a CnfStream that performs equisatisfiable CNF transformations
+   * and sends the generated clauses and to the given SAT solver. This does not
+   * take ownership of satSolver, registrar, or context.
+   *
+   * @param satSolver the sat solver to use.
+   * @param registrar the entity that takes care of preregistration of Nodes.
+   * @param context the context that the CNF should respect.
+   * @param outMgr Reference to the output manager of the smt engine. Assertions
+   * will not be dumped if outMgr == nullptr.
+   * @param fullLitToNodeMap maintain a full SAT-literal-to-Node mapping.
```
Comment (ajreynol) [inline src/prop/cnf_stream.h:72]: Resource manager is missing from doc.

File: src/prop/prop_engine.cpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ PropEngine::PropEngine(TheoryEngine* te,
   d_satSolver = SatSolverFactory::createDPLLMinisat(smtStatisticsRegistry());
 
   d_registrar = new theory::TheoryRegistrar(d_theoryEngine);
-  d_cnfStream = new CVC4::prop::TseitinCnfStream(
+  d_cnfStream = new CVC4::prop::CnfStream(
```
Comment (ajreynol) [inline src/prop/prop_engine.cpp:94]: Why this change (and the ones below?).  Was the derived class redundant?

File: src/prop/cnf_stream.h (hunk 1 above)
Comment (ajreynol) [reply src/prop/cnf_stream.h:72]: Also, I assume you are adding the resource manager because it will be used?

**Review by ajreynol — CHANGES_REQUESTED** [review]
Looks good, just some minor questions and comments.

File: src/prop/prop_engine.cpp (hunk 2 above)
Comment (HanielB) [reply src/prop/prop_engine.cpp:94]: The CnfStream was only ever used as a TseitingCnfStream extension. So I collapsed the latter into the former.

File: src/prop/cnf_stream.h (hunk 1 above)
Comment (HanielB) [reply src/prop/cnf_stream.h:72]: Will add. This was because it was used in the extension.

**Review by ajreynol — APPROVED** [review]
